### DISCUSSION TITLE
/hyndsight/forecast7-part-2/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#256] [Announcements]
Forecast v7 (part 2) | Rob J Hyndman 
 https://robjhyndman.com/hyndsight/forecast7-part-2/

**Comment by robjhyndman** [comment]
```header
Originally posted by Jason on 2016-06-08 00:12:17
````
This is amazing!! Thank you for all the hard work you put into the
package! A question about the covariate nnetar change, is this now an
alternative model to use instead of auto.arima with seasonal dummy
variables (i.e. floating holidays like Easter)?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2016-06-08 01:53:32
````
Yes

**Comment by robjhyndman** [comment]
```header
Originally posted by Mark Adamson on 2016-06-30 09:40:35
````
I\'ve updated our model to forecast v7 now, but it is now generating
warnings now when I call forecast.Arima that it didn\'t before with the
same call and same data. It seems to be due to a negative sigma\^2
coming from a call to Arima.

Warning:\
Error in sqrt(z\[\[2L\]\] \* object\$sigma2) :\
(converted from warning) NaNs produced

This happens within the predict function where it is calculating the
standard error I think (even though I asked for level=0):

se \&lt;- ts(sqrt(z\[\[2L\]\] \* object\$sigma2), start = xtsp\[2L\] +
deltat(rsd), frequency = xtsp\[3L\])

it is because sigma2 is negative in the model so the sqrt then fails. I
am using lambda=0.

My call to forecast is:

newModel \&lt;- Arima(x = hhReads, xreg = pastHHXReg, model = model)\
forecast.Arima(newModel, xreg = futureHHXReg, level = 0)

v6 of forecast produced a positive sigma2 for newModel, but v7 produces
a negative sigma2 so it seems that this is what has changed between the
two. sigma2 for model is positive in both.

**Reply by robjhyndman** [reply]
```header
Originally posted by Mark Adamson on 2016-06-30 09:49:19
````
.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2016-06-30 12:44:26
````
Please provide a reproducible example of the problem. Most likely the
model is mis-specified with rank-deficient regressors or
over-parameterized ARMA. The new version is picking up more problems
with models than before.

**Reply by robjhyndman** [reply]
```header
Originally posted by Mark Adamson on 2016-07-01 13:54:51
````
I\'ve now emailed you a reproduction of the problem. Hope it\'s ok for
your needs. Thanks, Mark

**Reply by robjhyndman** [reply]
```header
Originally posted on 2016-07-01 18:53:27
````
Thanks. I\'ve fixed the problem. You can grab the updated package from
[https://github.com/robjhynd\...](https://github.com/robjhyndman/forecast)

**Reply by robjhyndman** [reply]
```header
Originally posted by Mark Adamson on 2016-07-01 22:04:47
````
super, thanks for the quick turnaround!

**Comment by robjhyndman** [comment]
```header
Originally posted by Isabella Ghement on 2016-07-15 16:31:29
````
Dear Professor Hyndman,

Thank you for all of the improvements you have made to the forecast
package. It is an extremely valuable package for applied practitioners
of statistics and, unlike many other R packages, it has excellent
documentation via your articles, blog and online forecasting book.

I do have a question about the example provided in the help file for the
fitted.Arima() function. The help file states that the function
&quot;Returns one-step forecasts for the data used in fitting the ARIMA
model&quot;. If the original time series takes values x1, x2, x3, \... at
equally spaced times t1, t2, t3, \..., then are the fitted values
supposed to be displaced by one time unit relative to the original time
series values? In other words, when plotting the time series and fitted
values on the same plot, are we supposed to plot x2, x3, \...versus t2,
t3, \... and fitted1, fitted2, \... versus t2, t3, \...?\
Thank you in advance for your answer.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2016-07-15 17:45:27
````
No. The time stamps are correct on the fitted values. fitted1 is the
fitted value for t1, etc.

**Comment by robjhyndman** [comment]
```header
Originally posted by Elkin Tabares on 2016-07-27 03:10:06
````
Could I perfomance cross validation in the nnetar function?. Thank you
very much.

**Comment by robjhyndman** [comment]
```header
Originally posted by Ashin Mukherjee on 2016-08-22 04:07:28
````
Dear Prof. Hyndman,

Thanks for all the recent improvements to the forecast package. I had a
question about seasonal models in auto.arima. I was wondering if you
ever thought about adding an option to only search over seasonal models
in auto.arima. I understand that it is quite straightforward to restrict
the search space among non-seasonal models by setting the seasonal
parameter to FALSE but could not get an easy way of doing the reverse.
In my examples sometimes the data set is short maybe 3 to 4 years of
bi-weekly data and the seasonal pattern only became prominent in the
last couple of years so clearly not an ideal data set for box-jenkins
type of model. auto.arima selects a high order (3,1,1) ARIMA model as
the optimal choice which does fine in fitting but forecasts look rather
off due to the absence of seasonal orders. But when I tried
(0,1,1)x(0,1,1) the often quoted fall back model in econometric studies
I get much nicer looking forecasts at the cost of slight increase in
one-step ahead accuracy metrics. This example got me thinking whether it
would be possible to force a seasonal model in auto.arima similar to
what you have already done for non-seasonal models.

I look forward to hear from you.

Best regards,\
Ashin

**Reply by robjhyndman** [reply]
```header
Originally posted on 2016-08-23 00:28:25
````
I\'d rather not do that as it can lead to poor models. But you can force
D=1 to get seasonal differencing included. In general, if a seasonal
model is not being selected, there is a reason for it. The purpose in
setting seasonal=FALSE is to improve computational speed, not to
over-ride a good model.

**Reply by robjhyndman** [reply]
```header
Originally posted by Ashin Mukherjee on 2016-08-23 02:14:07
````
Thanks for the pointer I will try that out.

**Comment by robjhyndman** [comment]
```header
Originally posted by Rohan_Khatavkar on 2018-03-01 17:01:48
````
Hello Dr Hyndman,

My time series data has increasing trend. Is differencing available in
nnetar? I am using d=1 in the nnetar setting but my forecasts always
seem to be lower than actual.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-03-01 20:47:48
````
No, differencing is not part of the way nnetar works. You can always
difference your data manually using diff, and then apply nnetar, but you
will need to undo the differencing to use the forecasts.

**Reply by robjhyndman** [reply]
```header
Originally posted by Rohan_Khatavkar on 2018-03-05 17:03:10
````
Thank you for the response. I am using external regressors as well.\
1) Do I need to diff the Xreg matrix before using nnetar?\
2) Also I am using box cox transformation. This shouldn\'t affect the
back transformation process, correct? (I am using diffinv for back
transformation and setting initial value to last value of the training
series)

I get decent results with auto.arima but was exploring neural network to
see if I can improve.

**Comment by robjhyndman** [comment]
```header
Originally posted by nand on 2018-03-15 07:20:22
````
Hi Prof Rob, I have some 10 univariate time series with 80 observations
each (monthly data). I split the data as 72 for training period and 8
for testing period and use arima() and ets() to fit the training data
and forecast it and after considering the test data to find the mape and
choose the model. With the resulting model, I forecast 20 observations.
But many of series forecast results in constant values after 8-10 lags.
Could it be due to the lack of information in the series ( as the acf
peaks for stationary series are too small ) or could i get any better
model that gives better forecast using the package.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-03-15 10:13:35
````
Any stationary model will produce constant forecasts in the long run.

**Reply by robjhyndman** [reply]
```header
Originally posted by nand on 2018-03-15 14:42:40
````
Thanks. So for all those series having less number of observations(say
below 100) couldn\'t we forecast for 20 months ahead with ets, tbats or
any other package?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-03-15 21:34:15
````
Forecast with whatever method you think best. But if the series are
stationary, the forecasts should converge to the mean.

**Reply by robjhyndman** [reply]
```header
Originally posted by nand on 2018-03-16 09:49:04
````
Thank you very much. Does the forecast package is giving the forecast
values by considering the data to be stationary( so that the forecast
values becomes constant) Correct me if I am not. Could you please share
some reference to the same, so that we can use it forecast the series
more efficiently for long periods and also help in learning. Thank you
once more.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-03-17 01:55:21
````
Read my online book at
[otexts.org/fpp2/](http://otexts.org/fpp2/ &quot;otexts.org/fpp2/&quot;)

**Comment by robjhyndman** [comment]
```header
Originally posted on 2018-03-14 20:19:14
````
Different data sets can lead to different models. In fit1 you are
throwing away the time series meta data when you subset the data like
this. Using window preserves it.
